### PR TITLE
New Copyright proposal

### PR DESCRIPTION
**Pull request by permcody**

### CODE REVIEW

**Comment by permcody** [comment]
@idaholab/moose-team  - Please take a look at the proposed changes. If everyone agrees, we'll update all the headers. In fact, I would propose that we roll through the modules as well.

File: COPYRIGHT [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+Large portions of this software were developed at the Idaho National Laboratory
+
+Major contributors include the following:
+The Commonwealth Scientific and Industrial Research Organisation
+
+Other contributors include the following
+The University of Texas at Austin
+Argonne National Laboratory
+Los Alamos National Laboratory
+Oakridge National Laboratory
```
Comment (YaqiWang) [inline COPYRIGHT]: should be Oak Ridge. (ORNL)

File: COPYRIGHT [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+Massachusetts Institute of Technology
+
+For complete author information reference the Git repository
+https://github.com/idaholab/moose
+
+The following are the complete copyright notices for select instrutions
```
Comment (andrsd) [inline COPYRIGHT]: s/instrutions/institutions/

File: COPYRIGHT [hunk 3]
```diff
@@ -1,4 +1,21 @@
-© 2010 Battelle Energy Alliance, LLC
+Large portions of this software were developed at the Idaho National Laboratory
+
+Major contributors include the following:
+The Commonwealth Scientific and Industrial Research Organisation
+
+Other contributors include the following
```
Comment (andrsd) [inline COPYRIGHT]: `:` at the end

File: framework/include/base/Assembly.h [hunk 4]
```diff
@@ -2,14 +2,11 @@
 /*               DO NOT MODIFY THIS HEADER                      */
 /* MOOSE - Multiphysics Object Oriented Simulation Environment  */
 /*                                                              */
-/*           (c) 2010 Battelle Energy Alliance, LLC             */
-/*                   ALL RIGHTS RESERVED                        */
+/*   Copyrights reserved, Please see COPYRIGHT for details      */
+/*   https://github.com/idaholab/moose/blob/devel/COPYRIGHT     */
```
Comment (andrsd) [inline framework/include/base/Assembly.h]: Maybe `master` instead of `devel`?

**Review by andrsd — CHANGES_REQUESTED** [review]

File: framework/include/base/Assembly.h (hunk 4 above)
Comment (aeslaughter) [reply framework/include/base/Assembly.h]: Please should not be capital

**Comment by aeslaughter** [comment]
You will also need to look at the python headers, there are a few versions of it.

File: framework/include/base/Assembly.h [hunk 5]
```diff
@@ -2,14 +2,11 @@
 /*               DO NOT MODIFY THIS HEADER                      */
 /* MOOSE - Multiphysics Object Oriented Simulation Environment  */
 /*                                                              */
```
Comment (aeslaughter) [inline framework/include/base/Assembly.h]: I think we should change the width to be the 100 line length limit to make the file margins look consistent.

**Comment by permcody** [comment]
@idaholab/moose-team - New version pushed

**Review by andrsd — DISMISSED** [review]

File: COPYRIGHT [hunk 6]
```diff
@@ -1,4 +1,21 @@
...
+Los Alamos National Laboratory
+Oak Ridge National Laboratory
+Massachusetts Institute of Technology
+
+For complete author information reference the Git repository
+https://github.com/idaholab/moose
+
+The following are the complete copyright notices for select institutions
+
+© 2010-2018 Battelle Energy Alliance, LLC
```
Comment (jwpeterson) [inline COPYRIGHT]: Looks like this copyright symbol has always been a Unicode character. Not sure if we care about this/want to change it?

File: COPYRIGHT [hunk 7]
```diff
@@ -1,4 +1,21 @@
-© 2010 Battelle Energy Alliance, LLC
+Large portions of this software were developed at the Idaho National Laboratory
```
Comment (jwpeterson) [inline COPYRIGHT]: Period at end of line.

**Review by jwpeterson — DISMISSED** [review]
Are you not actually updating all the source files in this PR?

File: COPYRIGHT (hunk 6 above)
Comment (permcody) [reply COPYRIGHT]: Yeah, I'm not sure I really care about it either way unless you feel strongly about it.

**Comment by permcody** [comment]
So, I will update all of the headers, but not need to do that until I'm done tweaking the header.

File: framework/include/base/Assembly.h [hunk 8]
```diff
@@ -1,16 +1,13 @@
...
+/**************************************************************************************************/
+/*                                DO NOT MODIFY THIS HEADER                                       */
+/*                  MOOSE - Multiphysics Object Oriented Simulation Environment                   */
+/*                                                                                                */
+/*                    Copyrights reserved, please see COPYRIGHT for details                       */
+/*                    https://github.com/idaholab/moose/blob/master/COPYRIGHT                     */
+/*                                                                                                */
+/*                    Licensed under LGPL 2.1, please see LICENSE for details                     */
+/*                    https://www.gnu.org/licenses/lgpl-2.1.html                                  */
+/**************************************************************************************************/
```
Comment (friedmud) [inline framework/include/base/Assembly.h]: Why so wide?

Comment (rwcarlsen) [reply framework/include/base/Assembly.h]: I think this should be as narrow as possible to fit the text.

Comment (aeslaughter) [reply framework/include/base/Assembly.h]: I think it should fit to our defined line width, it makes the right margin match the code lines.

Comment (permcody) [reply framework/include/base/Assembly.h]: Honestly, I'm not a huge fan of the wide width either. I would venture that most headers aren't defined to be the width of the projects that they are on. The GPL headers are pretty short.

Comment (andrsd) [reply framework/include/base/Assembly.h]: So, drop the frame, align to the left and use `//` 😈

Comment (aeslaughter) [reply framework/include/base/Assembly.h]: I am with @andrsd

**Comment by friedmud** [comment]
I like the border (plus - we're used to looking at it... but let's make it the smallest we can make it (it will shave

File: framework/include/base/Assembly.h [hunk 9]
```diff
@@ -1,16 +1,13 @@
...
-/****************************************************************/
+/**************************************************************************************************/
+/*                                DO NOT MODIFY THIS HEADER                                       */
+/*                  MOOSE - Multiphysics Object Oriented Simulation Environment                   */
+/*                                                                                                */
+/*                    Copyrights reserved, please see COPYRIGHT for details                       */
+/*                    https://github.com/idaholab/moose/blob/master/COPYRIGHT                     */
+/*                                                                                                */
+/*                    Licensed under LGPL 2.1, please see LICENSE for details                     */
+/*                    https://www.gnu.org/licenses/lgpl-2.1.html                                  */
```
Comment (YaqiWang) [inline framework/include/base/Assembly.h]: We actually do not need these two links, they should both are included in COPYRIGHT and LICENSE files.

Comment (permcody) [reply framework/include/base/Assembly.h]: Except that these are here specifically for the case when files are separated from the repository (i.e. somebody mails a file to somebody else).

Comment (YaqiWang) [reply framework/include/base/Assembly.h]: oh. One more thing, is our LICENSE file the same as https://www.gnu.org/licenses/lgpl-2.1.html?

**Comment by rwcarlsen** [comment]
I am okay with either "//" or as narrow as possible.  Wide just looks
ridiculous.

**Comment by andrsd** [comment]
2 suggestions:

1. I would also remove the `DO NOT MODIFY` line. We did have people modifying it, but then all new code goes via review, so if they do modify it, we tell them "nope".

2. How about something like "This file is part of MOOSE framework (https://www.mooseframework.org)"

**Comment by friedmud** [comment]
I like the "This file is part of the MOOSE..." stuff - that's a good idea.

I still like the DO NOT MODIFY line.  Depending on what project you're used
to working on - some people might think they should modify that header.
Nothing wrong with having a little reminder there to leave the banner alone.

Derek

**Comment by rwcarlsen** [comment]
Couldn't we just have a CIVET check for modified headers? "do not modify"
warnings make more sense for generated full files/code where you might
modify something that you thought was regular code but was
generated/special.  I don't think any of us are in danger of intentionally
modifying the header.  And the warning text won't prevent unintentional
modifications.

**Comment by permcody** [comment]
I'm fine with @aeslaughters modifications with the exception of the missing periods at the end of each sentence. I'll add those in and push up a PR sometime later this week or early next week. Any last minute comments should be made today please.

**Comment by brianmoose** [comment]
Tutorials?

**Comment by permcody** [comment]
> Tutorials

Doh!

**Comment by permcody** [comment]
@brianmoose - This DOES include tutorials. I just double checked.

**Comment by brianmoose** [comment]
My bad, Civet will auto add the "Tutorials" target when there are changes in the tutorials but I didn't see it on this PR.  
It looks like GitHub will only return a max of 300 changed files via the API, which is why it didn't auto add it. I'll add it now.

**Review by friedmud — APPROVED** [review]
👍

**Pull request merged**